### ISSUE TITLE
Custom usergroups doesn't count as "loggedin users"

### ISSUE THREAD

**Issue by Maik0815** [nZEDb/nZEDb#2628]
I have added 2 additional usergroups and wrote an article (contents page) which should only be available for loggedin users. Users of default groups, like "role = 1, 2, 4" can see this article but users of group 6 and 7 can't see it, it only shows a blank contents page window.

I searched around files like content.php and content.tpl and didn't found something. I think these groups needs to be added to loggedin users role or something

**Comment by niel** [comment]
IIRC roles have no connection to logged-in status. Logged in, is checked separately.

**Comment by Maik0815** [comment]
Yes, logged-in and roles are seperated smh. I found something like that in the code. Do you have hints for me, how i can handle this ? Idk why new usergroups don't work, seems like that this isn't intended

**Comment by niel** [comment]
They're separate by design I think. It's the difference between authentication and authorisation. I'll try to find some time to take a look at this.
You'd be better joining us on the IRC channel for more help.

**Comment by niel** [comment]
I think I found the problem, but I do not see a simple fix for it.
The method that fetches the page content, checks if the content role is for everyone or for a specific user role. It's not really checking on the content role the way I would expect.

See the method at https://github.com/nZEDb/nZEDb/blob/0.x/nzedb/Contents.php#L220

I'll think about a good (enough) way to fix this.

**Comment by niel** [comment]
It's to complicated to fix this and not break content relying on current behaviour. It require re-architecting users and roles.

I am rewriting the entire web-UI currently and that re-architecting is part of the process I intend. However, that doesn't help you, as it will take a while.

You can replace lines 222-226 of the method mentioned above, with the following snippet to achieve what you want. Be aware that role (of the user) based content will not work as it does now though.

> $role = '';
> switch (true) {
> 	case $role == Users::ROLE_ADMIN || $role == Users::ROLE_MODERATOR:
> 		break;
> 	case $his->user->isLoggedIn():
> 		$role = ' AND (role < 2)';
> 	default:
> 		$role .= ' AND role = 0';
> }
